### PR TITLE
[DONT MERGE]Benchmark module added

### PR DESCRIPTION
**Pull request by xuyangit**
## What changes were proposed in this pull request?

Add benchmark module as oap-perf directory. 
Add shell scripts as oap-perf/bin/*.sh for basic workflow.

## How was this patch tested?

Cloned from repository and test the basic workflow from gen data to benchmark test on tpcds10. All steps work correctly.

(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

### CODE REVIEW

File: oap-perf/src/main/scala/BenchmarkTest.scala [hunk 1]
```diff
@@ -0,0 +1,206 @@
+/*
```
Comment (gczsjdy) [inline oap-perf/src/main/scala/BenchmarkTest.scala]: These should be in some package, maybe `org.apache.spark.sql.oap.perf`? I'm not sure.

File: oap-perf/README.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+## Prerequisites
+This module uses SBT-1.0.2 to build itself. Before running "sbt assembly" to build the module, please set the repositories mirrors or the downloading can be very slow.
+
+## Workflow
+* Firstly please run "sbt assembly" to build the necessary jar file for testing under the oap-perf directory.
+* Setting bin/benchmark.config for all settings of benchmark workflows.
+* Run bin/runDataGen.sh to generate tpcds data in hdfs.
+* Run bin/runCreateDatabase.sh to transform table (e.g, remove null values, scale values of index attributes) and generate table.
+* Run bin/runIndexBuild.sh to generate index.
+* Run bin/runBenchmarkTest.sh to generate benchmark results. Notice as we use clear cache commands in scala code, which needs sudo operation, you'd better run sudo bin/runBenchmarkTest.sh. Though the command won't throw execption if not executed as sudoers, there maybe impact on final results.
```
Comment (gczsjdy) [inline oap-perf/README.md]: According to experience in HiBench, we can provide just 2 scripts to simplify this, which is prepare and runBenchmark. We can still keep these scripts and call them in the 2 suggested ones.

File: oap-perf/bin/runBenchmarkTest.sh [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+# create database profiler;
+# eof
+
+# first source config
+. ./benchmark.config
+
+
+${sparkPath}/bin/spark-submit \
+	--class BenchmarkTest \
+	--master yarn \
```
Comment (gczsjdy) [inline oap-perf/bin/runBenchmarkTest.sh]: master should also be configurable? We can also extract a shell function for submit. Can you refer to https://github.com/intel-hadoop/HiBench/blob/833fbdcfcbf0a2e17c688c96e20883c7ca7db6c2/bin/functions/workload_functions.sh#L185 ?

File: oap-perf/README.md [hunk 4]
```diff
@@ -0,0 +1,15 @@
+# Benchmark guide of OAP
+This module serves as the benchmark test for OAP-0.3.0 features. 
+
+## Prerequisites
+This module uses SBT-1.0.2 to build itself. Before running "sbt assembly" to build the module, please set the repositories mirrors or the downloading can be very slow.
```
[the remark this answers is not in the document]
Comment (xuyangit) [reply oap-perf/README.md]: Yes. Could u help with this later?

File: oap-perf/README.md [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+
+## Workflow
+* Firstly please run "sbt assembly" to build the necessary jar file for testing under the oap-perf directory.
+* Setting bin/benchmark.config for all settings of benchmark workflows.
+* Run bin/runDataGen.sh to generate tpcds data in hdfs.
+* Run bin/runCreateDatabase.sh to transform table (e.g, remove null values, scale values of index attributes) and generate table.
```
[the remark this answers is not in the document]
Comment (xuyangit) [reply oap-perf/README.md]: ok, good advice

File: oap-perf/bin/runBenchmarkTest.sh (hunk 3 above)
Comment (xuyangit) [reply oap-perf/bin/runBenchmarkTest.sh]: ok, also move hdfs path out later.

File: oap-perf/bin/benchmark.config [hunk 6]
```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
```
[the remark this answers is not in the document]
Comment (xuyangit) [reply oap-perf/bin/benchmark.config]: As this is also script to some degrees, i am not sure about this.

File: oap-perf/README.md [hunk 7]
```diff
@@ -0,0 +1,15 @@
+# Benchmark guide of OAP
+This module serves as the benchmark test for OAP-0.3.0 features. 
+
+## Prerequisites
+This module uses SBT-1.0.2 to build itself. Before running "sbt assembly" to build the module, please set the repositories mirrors or the downloading can be very slow.
+
+## Workflow
+* Firstly please run "sbt assembly" to build the necessary jar file for testing under the oap-perf directory.
+* Setting bin/benchmark.config for all settings of benchmark workflows.
+* Run bin/runDataGen.sh to generate tpcds data in hdfs.
```
[the remark this answers is not in the document]
Comment (xuyangit) [reply oap-perf/README.md]: Fixed

File: oap-perf/README.md (hunk 2 above)
Comment (xuyangit) [reply oap-perf/README.md]: @gczsjdy ,get it

**Comment by linhong-intel** [comment]
@yueawang @xuyangit 
Should we close this?

**Comment by yueawang** [comment]
Yes, let's close it. Thanks!

**Pull request closed without merging**